### PR TITLE
Rework "more" links for outings and routes

### LINKED ISSUE #754: Sort associated route on waypoints, books...
**Community discussion about this feature (mandatory)**
https://forum.camptocamp.org/t/projet-ui-liste-des-itineraires-associes-a-un-point-de-passage/235962/76?u=charlesb

**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
On https://www.camptocamp.org/waypoints/37582/fr/mont-blanc-du-tacul, on associated routes block, sort items by title, instead of by id

**Describe alternatives you've considered**
Rating ? may be hard to choose one

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by brunobesson**
Assuming the API is fixed (no duplicates) and the limit is augmented for routes from 30 to 400

* sort routes alphabetically
* move more link to the bottom in a button with number of results in badge
Fixes #754 

⚠️ wait for https://github.com/c2corg/v6_api/issues?q=is%3Aopen+is%3Aissue+milestone%3Av6.8.0 release

### CODE REVIEW

File: src/views/document/utils/boxes/RecentOutingsBox.vue [hunk 1]
```diff
@@ -3,23 +3,25 @@
...
-      <add-link document-type="outing" :query="query" class="button is-primary">
-        <span v-if="outings.length === 0" v-translate> Add the first outing </span>
+    <div
+      v-if="(!hideSeeAllResultsButton && outings.length) || documentType == 'route'"
+      class="has-text-centered add-section"
+    >
+      <router-link
+        :to="{ name: 'outings', query: query }"
+        class="button is-primary"
+        v-if="outings.length && !hideSeeAllResultsButton"
```
Comment (ataraxido) [inline src/views/document/utils/boxes/RecentOutingsBox.vue:19]: should be "!hideSeeAllResultsButton && outings.length" for performance & consistency

File: src/views/document/utils/field-viewers/FieldView.vue [hunk 2]
```diff
@@ -47,7 +47,7 @@ export default {
         return false;
       }
 
-      if (Array.isArray(value) && value.length === 0) {
+      if (Array.isArray(value) && !value.length) {
```
Comment (ataraxido) [inline src/views/document/utils/field-viewers/FieldView.vue:50]: previous version seems better

File: src/views/document/utils/boxes/RecentOutingsBox.vue [hunk 3]
```diff
@@ -41,23 +43,13 @@ export default {
...
-        outings = associations.recent_outings.documents;
-      } else {
-        outings = associations.outings;
-      }
-
-      for (const outing of outings) {
-        result.set(outing.document_id, outing);
-      }
+      const outings = this.document.associations.recent_outings?.documents || this.document.associations.outings;
+      return outings.filter((outing) => outing.quality !== 'empty');
```
Comment (ataraxido) [inline src/views/document/utils/boxes/RecentOutingsBox.vue:48]: could be combined like below ( RoutesBox.vue l55 )

**Review by ataraxido — CHANGES_REQUESTED** [review]

File: src/views/document/utils/field-viewers/FieldView.vue (hunk 2 above)
Comment (brunobesson) [reply src/views/document/utils/field-viewers/FieldView.vue:50]: I totally disagree 🙂 
Ok, it's mainly a matter of taste. But there are also more subtle reasons for this IMHO, among which the fact that any falsy value will be tested (think what if `length` was `undefined`? OK again, here we test for array first)

And you save 5 characters, it's like saving baby seals 👍

File: src/views/document/utils/boxes/RecentOutingsBox.vue (hunk 1 above)
Comment (brunobesson) [reply src/views/document/utils/boxes/RecentOutingsBox.vue:19]: I don't think retrieving an array's length has a bigger cost than checking a boolean? It's a simple address lookup. OK for consistency of reading though.

File: src/views/document/utils/field-viewers/FieldView.vue (hunk 2 above)
Comment (ataraxido) [reply src/views/document/utils/field-viewers/FieldView.vue:50]: Brigitte Bardot approves

**Review by ataraxido — APPROVED** [review]

**Pull request merged**